### ISSUE TITLE
pyOpenMS builds failing

### ISSUE THREAD

**Issue by hroest** [OpenMS/OpenMS#1893]
currently, the pyOpenMS builds are failing without error message on cdash: 

http://cdash.openms.de/index.php?project=OpenMS&date=20160329

basically no warning, no errors but then
```
======================================================================
ERROR: Failure: ImportError (No module named pyopenms)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/Library/Python/2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/Library/Python/2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/Users/aiche/NIGHTLY/builds/mac-osx-10.8-clang-5.0-LLVM3.3svn-Unix_Makefiles-Release/pyOpenMS/tests/unittests/test000.py", line 3, in <module>
    import pyopenms
  File "/Users/aiche/NIGHTLY/builds/mac-osx-10.8-clang-5.0-LLVM3.3svn-Unix_Makefiles-Release/pyOpenMS/pyopenms/__init__.py", line 52, in <module>
    raise e
ImportError: No module named pyopenms
```

any idea what is going wrong and why no warnings show up?

**Comment by jpfeuffer** [comment]
You are right. I was actually quite sure that my PR fixed the pyopenms issue. I worked on the build machine, so the environment was the same. I assume it is again a permission/overriding/uploading error.
I am sorry for the build environment being a bit unstable in the last weeks. I am focussing on setting up a new Jenkins-based one where more people have access to and where we have more possibilities.
Nonetheless I am looking into it right now.

**Comment by jpfeuffer** [comment]
Argh.. sure.. we pushed my fix it to master and it is building develop..

**Comment by jpfeuffer** [comment]
Okay. The error changed:
http://cdash.openms.de/viewBuildError.php?buildid=172087

Where do I have to look to see why there is unreachable code? Did you experience such errors before?

**Comment by hendrikweisser** [comment]
I'm pretty sure you can ignore the "unreachable code" warnings. They shouldn't lead to a build failure.
Looks to me like the problem is with QcMLFile.

**Comment by jpfeuffer** [comment]
Okay, yes makes sense, they are only warnings. Another related question:
The PyChecker seems like it is also not functioning correctly. It reports missing pxds although they are there in the sources.
http://cdash.openms.de/viewTest.php?onlyfailed&buildid=172090
Where does it look for the pxds?

**Comment by jpfeuffer** [comment]
Problem is that the script is very sensitive to where to call it from and how to correctly specify the locations of source and build.
All I know is, that the mapping seems to be wrong in the end:
It reads the .h files from the pxds like this:
['OpenMS/CHEMISTRY/AAIndex.h']
But then looks for them in the map later with the full path:
/Users/aiche/NIGHTLY/RELEASE/OpenMS/CHEMISTRY/AAIndex.h

I don't know. I'll wait for @hroest  input on this.

**Comment by hroest** [comment]
@jpfeuffer the script has a `--bin_path` and a `--src_path` argument from which it tries to derive this information, however the trouble seems to appear here:

https://github.com/OpenMS/OpenMS/blob/9a795431d0e8f45bc9a4a6462f3b83517b72e9b3/tools/PythonExtensionChecker.py#L1033 : 


```
dfile = DoxygenXMLFile(f)
file_location = dfile.getCompoundFileLocation()

file_location_key = file_location
if len(file_location.split("include/")) > 1: 
    file_location_key = file_location.split("include/")[1]

if file_location_key in pxd_file_matching:
    pxdfiles = pxd_file_matching[file_location_key]
else:
    Error
        
```
basically, we are given a file path from the XML file and we need to figure out which part of this file path is our "key" that we stored earlier in the `pxd_file_matching` hash in (see `create_pxd_file_map` in https://github.com/OpenMS/OpenMS/blob/9a795431d0e8f45bc9a4a6462f3b83517b72e9b3/tools/PythonCheckerLib.py#L89 )

Now, for some reason, file_location is set to `/Users/aiche/NIGHTLY/RELEASE/OpenMS/METADATA/Acquisition.h` and there is no `include` in there any more, so splitting does not happen and we cannot extract the relevant part of the path (`OpenMS/METADATA/Acquisition.h`). Now we could of course try to parse the new path and I have thought about splitting at `OpenMS` but this may break other setups. But the best course of action may actually be to do that.

**Comment by hroest** [comment]
@jpfeuffer can you check whether #1910 fixes the issue?

**Comment by jpfeuffer** [comment]
Thanks a lot! I will.

**Comment by jpfeuffer** [comment]
I tested it and with a small additional fixed line it works. Was there a  change in locations recently, such that the include subfolder is not necessarily present anymore?

**Comment by hroest** [comment]
Not that I am aware of but it must have changed somehow.

H

sent from my smartphone -- you may keep any typos you find.

**Comment by jpfeuffer** [comment]
Okay. This part is fixed. But there is still this weird QCML problem.
http://cdash.openms.de/viewBuildError.php?buildid=172350

`INFO:root:attach to QcMLFile: errorString`
 Anyone has an idea on this?

**Comment by hroest** [comment]
that has been there from the beginning, a false positive from cdash ...

**Comment by jpfeuffer** [comment]
Ah perfect, in some way. Is it because the line contains *error*? Were there some thoughts on how to fix this at some point?
I would like to reduce CDash errors a bit (at least for the Build column) :)

**Comment by hroest** [comment]
We could try to reduce the output by autowrap. Or look into how to remove
false positive hits in cdash. Maybe there is a mechanism?

sent from my smartphone -- you may keep any typos you find.

**Comment by jpfeuffer** [comment]
Okay, it seems it can be suppressed by introducing excpetions in the CTEST_CUSTOM_ERROR_EXCEPTION variable.
I'll se what I can do with this in the test scripts.

**Comment by hroest** [comment]
Sounds good!

sent from my smartphone -- you may keep any typos you find.

**Comment by jpfeuffer** [comment]
Small update: I found out that we already have a CMake file that is supposed to do exactly that but it is currently not used/accepted by CTest somehow.

**Comment by jpfeuffer** [comment]
Resolved with #1911 #1884 #1885

**Comment by hroest** [comment]
great :+1:
